std/traits.d

```diff
@@ -5315,12 +5325,19 @@ enum bool isIntegral(T) = is(IntegralTypeOf!T) && !isAggregateType!T;
     enum EI : int { a = -1, b = 0, c = 1 }  // base type is signed (bug 7909)
     static assert(isIntegral!EU &&  isUnsigned!EU && !isSigned!EU);
     static assert(isIntegral!EI && !isUnsigned!EI &&  isSigned!EI);
+
+    static struct S(T)
+    {
+        T t;
+        alias t this;
+    }
+    static assert(!isIntegral!(S!int));
 }
 
 /**
  * Detect whether $(D T) is a built-in floating point type.
  */
-enum bool isFloatingPoint(T) = is(FloatingPointTypeOf!T) && !isAggregateType!T;
+enum bool isFloatingPoint(T) = __traits(isFloating, T);
```

Comment:
This caused a semantic change which broke [scid-0.3.0](https://github.com/DlangScience/scid)'s testsuite, see [Issue 17195 – [Reg 2.074] isFloatingPoint!cfloat is now true](https://issues.dlang.org/show_bug.cgi?id=17195) @nordlow.